sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -452,32 +452,31 @@ def setup_demod_map(noise_model, shape=None, wcs=None, nside=None,
                                          singlestream=singlestream)
     return mapmaker
 
-def write_demod_maps(prefix, data, info, unit='K', split_labels=None, atomic_db=None):
+def write_demod_maps(prefix, data, info, unit='K', split_labels=None):
     """
     Write maps from data into files
     """
     Nsplits = len(split_labels)
     for n_split in range(Nsplits):
         if np.all(data.wmap[n_split] == 0.0):
-            if atomic_db is not None:
-                smutils.atomic_db_aux(atomic_db, info[n_split], valid=False)
+            info[n_split]['valid'] = False
             continue
+        else:
+            info[n_split]['valid'] = True
         data.signal.write(prefix, "%s_wmap"%split_labels[n_split],
```

Comment:
Since you want to write the data signals if `info[n_split]['valid'] = True` why don't you push them a level in?

That would make the continue (line 463) unnecessary, and make the code a bit easier to read.